DNN Platform/DotNetNuke.Web/Api/DnnAuthorizeAttribute.cs

```diff
@@ -89,20 +115,32 @@ public override bool IsAuthorized(AuthFilterContext context)
                 }
             }
 
+            // if the attribute opted in explicitly for specific authentication types, then 
+            // use it; otherwise use the defaults according to settings in the web.config.
+            var currentAuthType = (identity.AuthenticationType ?? "").Trim();
+            if (currentAuthType.Length > 0)
+            {
+                if (_authTypesSplit.Any())
+                {
+                    return _authTypesSplit.Contains(currentAuthType);
+                }
+
+                return DefaultAuthTypes.Contains(currentAuthType);
+            }
+
             return true;
         }
 
-        private string[] SplitString(string original)
+        private static readonly string[] EmptyArray = new string[0];
+        private static string[] SplitString(string original)
         {
-            if (String.IsNullOrEmpty(original))
+            if (string.IsNullOrEmpty(original))
             {
-                return new string[0];
+                return EmptyArray;
             }
 
-            IEnumerable<string> split = from piece in original.Split(',')
-                                        let trimmed = piece.Trim()
-                                        where !String.IsNullOrEmpty(trimmed)
-                                        select trimmed;
+            var split = from piece in original.Split(new[] {','}, StringSplitOptions.RemoveEmptyEntries)
+                        select piece.Trim();
```

Comment:
This change introduces the possibility of values which are empty after trimming, is that intended?